File: message-service/src/main/java/net/wojtek/zti/models/response/MessageResponse.java
package net.wojtek.zti.models.response;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import net.wojtek.zti.dao.MessageRepository;
import net.wojtek.zti.models.Message;

import java.time.LocalDate;
import java.time.LocalDateTime;

@NoArgsConstructor
@Getter
@Setter
public class MessageResponse {
    private int id;
    private int senderId;
    private int receiverId;
    private String content;
    private LocalDateTime date;

    public MessageResponse(Message message){
        this.id = message.getId();
        this.senderId = message.getSender().getId();
        this.receiverId = message.getReceiver().getId();
        this.content = message.getContent();
        this.date = message.getDate();
    }
}

File: gateway/Models/Responses/MessageResponse.cs
using Newtonsoft.Json;

namespace Gateway.Models.Responses;

public class MessageResponse
{
    [JsonProperty("id")]
    public int Id{ get; set; }
    [JsonProperty("receiverId")]
    public int ReceiverId{ get; set; }
    [JsonProperty("senderId")]
    public int SenderId{ get; set; }
    [JsonProperty("content")]
    public String Content{ get; set; } = null!;
    [JsonProperty("date")]
    public DateTime Date{ get; set; }
}